build/source/engine/canopySnow.f90

```diff
@@ -158,86 +170,66 @@ subroutine canopySnow(&
 
  ! get a trial value for canopy storage
  scalarCanopyIceIter = scalarCanopyIce
-
- ! iterate
  do iter=1,maxiter
-
-  ! ** compute unloading
-  scalarCanopySnowUnloading = snowUnloadingCoeff*scalarCanopyIceIter
-  unloadingDeriv            = snowUnloadingCoeff
-
-  ! ** compute throughfall
-
-  ! no snowfall
-  if(scalarSnowfall<tiny(dt))then ! no snow
-   ! compute throughfall -- note this is effectively zero (no snow case)
-   scalarThroughfallSnow = scalarSnowfall  ! throughfall (kg m-2 s-1)
-   canopyIceScaleFactor  = valueMissing    ! not used
-   throughfallDeriv      = 0._dp
-
-  ! snowfall: compute interception
-  else
-
-   ! ** process different options for maximum branch snow interception
-   select case(ixSnowInterception)
-
-    ! * option 1: maximum interception capacity an inverse function of new snow density (e.g., Mahat and Tarboton, HydProc 2013)
-    case(lightSnow)
-     ! (check new snow density is valid)
-     if(scalarNewSnowDensity < 0._dp)then; err=20; message=trim(message)//'invalid new snow density'; return; end if
-     ! (compute storage capacity of new snow)
-     leafScaleFactor       = 0.27_dp + 46._dp/scalarNewSnowDensity
-     leafInterceptCapSnow  = refInterceptCapSnow*leafScaleFactor  ! per unit leaf area (kg m-2)
-
-    ! * option 2: maximum interception capacity an increasing function of air temerature
-    case(stickySnow)
-     airtemp_degC = scalarAirtemp - Tfreeze
-     if    (airtemp_degC > -1._dp)then; leafScaleFactor = 4.0_dp
-     elseif(airtemp_degC > -3._dp)then; leafScaleFactor = 1.5_dp*airtemp_degC + 5.5_dp
-                                  else; leafScaleFactor = 1.0_dp
+     ! ** compute unloading
+     if (ixSnowUnload==meltDripUnload) then
+         scalarCanopySnowUnloading = snowUnloadingCoeff*scalarCanopyIceIter
+         unloadingDeriv            = snowUnloadingCoeff
+     else if (ixSnowUnload==windUnload) then
+         tempUnloadingFun = (C_1 + scalarCanairTemp) / C_2   ! (s-1)
+         windUnloadingFun = abs(scalarWindspdCanopyTop) / C_3     ! (s-1)
+         ! No snow unloading if T < -3C
```

Comment:
Comment does not appear to match the line that follows and is confusing `<-3` vs `<0` in the following line. Should this be a hard-coded constant or something that should be set in a parameter file?

Reply:
This has to do with the `C_1 + scalarCanairTemp` portion in `tempUnloadingFun`. That part evaluates to `scalarCanairTemp - 270.15`, hence the 3C. It seems like a better approach may be to be set in the parameter file, along with `C_2` and `C_3`. Along with better names, of course.